src/Microsoft.DotNet.Interactive.Tests/VariableSharingTests.cs

```diff
@@ -117,6 +117,92 @@ public class VariableSharingTests
                   .Be("123:System.Int32");
         }
 
+        [Theory]
```

Comment:
It's not clear how these are different from the existing tests.

Reply:
Instead of simply repeating the value, an operation is performed on it, e.g., `x + 1`.  This ensures that the type of `x` and the type of `1` refer to the same `System.Int32`.

Reply:
Why not change the existing tests rather than duplicate? Or make it clear that these tests are also verifying that the type is correct?

Reply:
made test names more explicit